File: py/0069.py
class Solution:
    def mySqrt(self, x: int) -> int:
        res = x/2
        if x == 0 or x == 1:
            return x
        for i in range(1000):
            if abs(res*res-x<1e-4):
                return int(res)
            res = 0.5*(res + x/res)
        return res

File: cpp/0069.cpp
//https://cloud.tencent.com/developer/article/1539037

class Solution
{
public:
    // 牛顿法
    int mySqrt(int x)
    {
        double res = 3;
        int n = 1000; //迭代次数
        for (int i = 0; i < n; i++)
        {
            if (abs(res * res - x) <= 1e-4)
            {
                return int(res);
            }
            res = 0.5 * (res + x / res);
        }
        return int(res);
    }
};